### ISSUE TITLE
no message when `newdata` ignored by `augment.kmeans()`

### ISSUE THREAD

**Issue by brshallo** [tidymodels/broom#1043]
`newdata` when running `augment()` on a kmeans object seems like it is being ignored though does not output a message telling the user as such (though an error is returned if `data` is not provided) (seems related to #450).

For example: 

``` r
library(palmerpenguins)
library(dplyr)
library(rsample)
library(recipes)

penguins <- na.omit(penguins)

set.seed(2)
split <- initial_split(penguins)
train <- training(split)
test <- testing(split)

rec_prepped <- recipe(~., data = train) %>% 
  step_normalize(contains("_")) %>% 
  prep()

train_baked <- juice(rec_prepped)
test_baked <- bake(rec_prepped, test)

kclusts <- kmeans(select(train_baked, contains("_")), 
                  centers = 4)

dim(train_baked)
#> [1] 249   8
dim(test_baked)
#> [1] 84  8

# based on row numbers see is adding cols onto train
broom::augment(kclusts, 
               data = train_baked, 
               newdata = test_baked)
#> # A tibble: 249 × 9
#>    species   island    bill_length_mm bill_depth_mm flipper_length_… body_mass_g
#>    <fct>     <fct>              <dbl>         <dbl>            <dbl>       <dbl>
#>  1 Gentoo    Biscoe             0.195       -1.39            0.652        0.222 
#>  2 Gentoo    Biscoe             0.513       -1.44            1.02         0.786 
#>  3 Chinstrap Dream              1.35         0.545          -0.294       -0.594 
#>  4 Gentoo    Biscoe            -0.160       -1.34            0.507        0.284 
#>  5 Chinstrap Dream              1.90         1.87           -0.00263      0.0963
#>  6 Gentoo    Biscoe             0.195       -1.34            0.434        1.04  
#>  7 Adelie    Torgersen         -1.77         0.0367         -0.875       -1.28  
#>  8 Adelie    Dream             -1.58         0.189          -0.730       -1.32  
#>  9 Chinstrap Dream              0.924        1.21            0.652       -0.343 
#> 10 Chinstrap Dream              1.26         0.697          -0.00263      0.284 
#> # … with 239 more rows, and 3 more variables: sex <fct>, year <int>,
#> #   .cluster <fct>
```

<sup>Created on 2021-08-19 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessionInfo()
#> R version 4.1.0 (2021-05-18)
#> Platform: x86_64-pc-linux-gnu (64-bit)
#> Running under: Ubuntu 20.04.2 LTS
#> 
#> Matrix products: default
#> BLAS:   /usr/lib/x86_64-linux-gnu/atlas/libblas.so.3.10.3
#> LAPACK: /usr/lib/x86_64-linux-gnu/atlas/liblapack.so.3.10.3
#> 
#> locale:
#>  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
#>  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
#>  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
#> [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> other attached packages:
#> [1] recipes_0.1.16       rsample_0.1.0        dplyr_1.0.7         
#> [4] palmerpenguins_0.1.0
#> 
#> loaded via a namespace (and not attached):
#>  [1] tidyselect_1.1.1   xfun_0.25          purrr_0.3.4        listenv_0.8.0     
#>  [5] splines_4.1.0      lattice_0.20-44    vctrs_0.3.8        generics_0.1.0    
#>  [9] htmltools_0.5.1.1  yaml_2.2.1         utf8_1.2.2         survival_3.2-11   
#> [13] prodlim_2019.11.13 rlang_0.4.11       pillar_1.6.2       glue_1.4.2        
#> [17] withr_2.4.2        lifecycle_1.0.0    lava_1.6.9         stringr_1.4.0     
#> [21] timeDate_3043.102  future_1.21.0      codetools_0.2-18   evaluate_0.14     
#> [25] knitr_1.33         parallel_4.1.0     class_7.3-19       fansi_0.5.0       
#> [29] highr_0.9          broom_0.7.9.9000   furrr_0.2.3        Rcpp_1.0.7        
#> [33] backports_1.2.1    ipred_0.9-11       parallelly_1.27.0  fs_1.5.0          
#> [37] digest_0.6.27      stringi_1.7.3      grid_4.1.0         cli_3.0.1         
#> [41] tools_4.1.0        magrittr_2.0.1     tibble_3.1.3       crayon_1.4.1      
#> [45] tidyr_1.1.3        pkgconfig_2.0.3    ellipsis_0.3.2     MASS_7.3-54       
#> [49] Matrix_1.3-3       reprex_2.0.1       lubridate_1.7.10   gower_0.2.2       
#> [53] rmarkdown_2.10     rstudioapi_0.13    R6_2.5.1           globals_0.14.0    
#> [57] rpart_4.1-15       nnet_7.3-16        compiler_4.1.0
```

</details>

P.s. would be cool if `augment.kmeans()` could take in `newdata` so could review cluster assignment on a holdout dataset (like rough solution here: https://stackoverflow.com/a/68845111/9059865) . Though seems like  is being discussed for doing elsewhere https://github.com/tidymodels/embed/issues/77#issuecomment-801300749  so maybe out of scope for broom.

**Comment by simonpcouch** [comment]
Thanks for the issue! `augment.kmeans` will continue not to take a `newdata` argument explicitly, though I agree that silently ignoring the `newdata` argument passed via `...` should be addressed. See #413 for related discussion.

**Comment by ddsjoberg** [comment]
Hello!

I wanted to drop a quick note here about using `rlang::check_dots_used()` to check for unused arguments. I use broom functions _all_ the time and I love them. I also included them in a package I wrote called gtsummary for prepping regression model results for publication. In the gtsummary package, I will always pass arguments `conf.int=`, `conf.level=`, and `exponentiate=` to every tidier. I know there are some regression tidiers that don't accept these args (e.g. the GAM tidier was just recently updated to accept the `exponentiate=` arg). Rather than failing if a user passes an bad arg, can users be signaled an argument was ignored, e.g. `"the exponentitate argument was not used"`? I believe that `rlang::check_dots_used()` will throw an error in this situation.

Thanks!
